src/Sarif/Visitors/UpdateIndicesVisitor.cs

```diff
@@ -27,7 +27,7 @@ public override Result VisitResult(Result node)
         {
             if (_ruleKeyToIndexMap != null)
             {
-                if (_ruleKeyToIndexMap.TryGetValue(node.RuleId, out int ruleIndex))
+                if (node.RuleId != null &&_ruleKeyToIndexMap.TryGetValue(node.RuleId, out int ruleIndex))
```

Comment:
>node.RuleId [](start = 20, length = 11)

One of the test cases has a result with a `ruleKey` but no `ruleId`. That is valid SARIF v1, and we need to protect against it.